packages/core/src/utils/setup-package-json.ts

```diff
@@ -44,64 +49,55 @@ export const diffPackageJson = (
   diffs: Difference[];
   expectedPackageJson: PackageJson;
 } => {
-  const files = new Set<string>(packageJson.files ?? []);
+  const existingFiles = new Set(packageJson.files ?? []);
+  const diffs: Difference[] = [];
+
+  const expectedPackageJson = sortPackageJson(structuredClone(packageJson));
 
-  let main: PackageJson['main'];
-  let module: PackageJson['module'];
+  expectedPackageJson.exports = getExportsForPackage(entries);
 
+  const files = new Set(existingFiles);
   for (const entryPoint of entries) {
     if (entryPoint.isDefaultEntry) {
-      main = `./${entryPoint.getOutputPath('cjs')}`;
-      module = `./${entryPoint.getOutputPath('esm')}`;
+      expectedPackageJson.main = `./${entryPoint.getOutputPath('cjs')}`;
+      expectedPackageJson.module = `./${entryPoint.getOutputPath('esm')}`;
     }
-
-    files.add(`/${entryPoint.entryName}`);
+    files.add(entryPoint.entryName);
   }
+  expectedPackageJson.files = sortFiles(files);
 
-  const exports = getExportsForPackage(entries);
-
-  const filesArray = Array.from(files);
-  const diffs: Difference[] = [];
-
-  if (main !== packageJson.main) {
+  if (expectedPackageJson.main !== packageJson.main) {
     diffs.push({
       key: 'main',
       from: packageJson.main,
-      to: main,
+      to: expectedPackageJson.main,
     });
   }
 
-  if (module !== packageJson.module) {
+  if (expectedPackageJson.module !== packageJson.module) {
     diffs.push({
       key: 'module',
       from: packageJson.module,
-      to: module,
+      to: expectedPackageJson.module,
     });
   }
 
-  if (files.size !== packageJson.files?.length) {
-    const missingFiles = filesArray.filter(
-      (file) => !packageJson.files?.includes(file),
+  if (!isDeepStrictEqual(expectedPackageJson.files, packageJson.files)) {
+    const missingFiles = expectedPackageJson.files.filter(
+      (file) => !existingFiles.has(file),
     );
     diffs.push({ key: 'files', additions: missingFiles });
   }
 
-  const packageJsonExports = packageJson.exports;
-
-  if (!isDeepStrictEqual(packageJsonExports, exports)) {
+  if (
+    !isDeepStrictEqual(
+      Object.entries(expectedPackageJson.exports!),
```

Comment:
Could this `!` be removed by defaulting to `{}`, just like `packageJson.exports`?
```suggestion
      Object.entries(expectedPackageJson.exports || {}),
```

Reply:
See line 57